src/Avalonia.Controls/TreeView.cs

```diff
@@ -353,7 +355,7 @@ private void SelectedItemsAdded(IList items)
                 MarkItemSelected(item, true);
             }
 
-            if (SelectedItem == null && !_syncingSelectedItems)
```

Comment:
I don't think removing this check is correct. Elsewhere, if you have a multi-selecting control and select one item followed by another then `SelectedItem` remains as the first selected item, i.e. `SelectedItem` represents the first selected item in a multi-selection and not the last. I think this is the case in WPF as well.

Reply:
I think the solution would be to surround the whole of `SelectSingleItem` in a `_syncingSelectedItems` fence and manually `SetAndRaise` the `SelectedItem` property.

Reply:
Which would also mean the `_syncingSelectedItems` fence in `SelectedItem.set` is unnecessary.